view/src/service.rs

```diff
@@ -223,6 +234,31 @@ impl ViewProtocol for ViewService {
         Box<dyn futures::Stream<Item = Result<pb::StatusStreamResponse, tonic::Status>> + Send>,
     >;
 
+    async fn await_change(
+        &self,
+        request: tonic::Request<pb::AwaitChangeRequest>,
+    ) -> Result<tonic::Response<pb::NoteRecord>, tonic::Status> {
+        self.check_worker().await?;
+        self.check_fvk(request.get_ref().fvk_hash.as_ref()).await?;
+
+        let note_commitment = request.get_ref().note_commitment.as_ref();
```

Comment:
This might be cleaner with something like into_inner(), then ok_or_else(|| tonic::Status ... )? to get rid of the option, then try_into() to convert to the domain type. Then you just have the note commitment domain type and can pass it directly to the storage method